src/grpc_server.rs

```diff
@@ -0,0 +1,253 @@
+        Ok(Response::new(response))
+    }
+
+    /// # Delete Deployment
+    /// Delete a configuration to a deployment from its uuid.
+    /// # Arguments
+    /// The request containing the uuid of the deployment to delete.
+    /// # Returns
+    /// Nothing, wrapped in a Result.
+    async fn delete_deployment(
+        &self,
+        request: Request<DeleteDeploymentRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let deployment_uuid = request.into_inner().deployment_uuid;
+
+        let app = get_app(&self.db, deployment_uuid);
+
+        let message: String = match self
+            .docker_engine
+            .stop_application(app.project_id.as_str(), app.application_id.as_str())
+            .await
+        {
+            Ok(_) => {
+                //TODO: change message when deletion is implemented
+                format!("Stopped application {}. It was not deleted", app.project_id)
+            }
+            Err(e) => {
+                format!("Failed to delete application {}: {}", app.project_id, e)
+            }
+        };
+
+        //TODO: Delete the app from the database & prune its image
+
+        let response = ResponseMessage { message };
+        Ok(Response::new(response))
+    }
+
+    /// # Stop Deployment
+    /// Stop a deployment from its uuid.
+    /// # Arguments
+    /// The request containing the uuid of the deployment to stop.
+    /// # Returns
+    /// Nothing, wrapped in a Result.
+    async fn stop_deployment(
+        &self,
+        request: Request<StopDeploymentRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let deployment_uuid = request.into_inner().deployment_uuid;
+
+        let app = get_app(&self.db, deployment_uuid);
+
+        let message: String = match self
+            .docker_engine
+            .stop_application(app.project_id.as_str(), app.application_id.as_str())
+            .await
+        {
+            Ok(_) => {
+                format!("Stopped application {}", app.project_id)
+            }
+            Err(e) => {
+                format!("Failed to stop application {}: {}", app.project_id, e)
+            }
+        };
+
+        let response = ResponseMessage { message };
+        Ok(Response::new(response))
+    }
+
+    /// # Deployment Status
+    /// Get the status of a deployment from its uuid.
+    /// # Arguments
+    /// The request containing the uuid of the deployment to get the status of.
+    /// # Returns
+    /// Nothing, wrapped in a Result.
+    async fn deployment_status(
+        &self,
+        request: Request<DeploymentStatusRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let deployment_uuid = request.into_inner().deployment_uuid;
+
+        let app = get_app(&self.db, deployment_uuid);
+
+        let status = self
+            .docker_engine
+            .get_application_status(app.project_id.as_str(), app.application_id.as_str())
+            .await
+            .unwrap();
```

Comment:
:fearful:?

Reply:
🤔⁉️

Reply:
`unwrap()` cause the program to panic (inside a thread I think it's just the thread that stops)

I don't see the use for it here